substrate/client/rpc-spec-v2/src/archive/mod.rs

```diff
@@ -33,3 +33,76 @@ pub mod error;
 
 pub use api::ArchiveApiServer;
 pub use archive::Archive;
+
+use serde::{Serialize, Deserialize};
+
+/// The result of an RPC method.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[serde(untagged)]
+pub enum MethodResult {
+	/// Method generated a result.
+	Ok(MethodResultOk),
+	/// Method encountered an error.
+	Err(MethodResultErr),
+}
+
+impl MethodResult {
+	/// Constructs a successful result.
+	pub(crate) fn ok(result: impl Into<String>) -> MethodResult {
```

Comment:
sorry I missed that MethodResult was used in the public API.

Thus, please revert this `pub(crate)` here